src/lib/tabwidget/tabwidget.cpp

```diff
@@ -695,6 +695,11 @@ bool TabWidget::canRestoreTab() const
     return m_closedTabsManager->isClosedTabAvailable();
 }
 
+bool TabWidget::isCurrentTabNew()
+{
+    return m_urlOnNewTab == weTab()->url();
```

Comment:
Umm, no. This is just wrong, you will treat *any* current tab with loaded m_urlOnNewTab as new tab.